File: src/api/Models/Course.cs
namespace Shadowbrook.Api.Models;

public class Course
{
    public Guid Id { get; set; }
    public required string Name { get; set; }
    public string? StreetAddress { get; set; }
    public string? City { get; set; }
    public string? State { get; set; }
    public string? ZipCode { get; set; }
    public string? ContactEmail { get; set; }
    public string? ContactPhone { get; set; }
    public int? TeeTimeIntervalMinutes { get; set; }
    public TimeOnly? FirstTeeTime { get; set; }
    public TimeOnly? LastTeeTime { get; set; }
    public decimal? FlatRatePrice { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }
}

File: src/web/src/types/course.ts
export interface Course {
  id: string;
  name: string;
  tenantId: string;
  tenantName?: string;
  timeZoneId: string;
  streetAddress?: string;
  city?: string;
  state?: string;
  zipCode?: string;
  contactEmail?: string;
  contactPhone?: string;
  createdAt: string;
  updatedAt: string;
}

export interface TeeTimeSettings {
  teeTimeIntervalMinutes: number;
  firstTeeTime: string;
  lastTeeTime: string;
  defaultCapacity: number;
}

export interface Pricing {
  defaultPrice: number | null;
  minPrice: number | null;
  maxPrice: number | null;
  schedules: RateSchedule[];
}

export interface RateSchedule {
  id: string;
  name: string;
  daysOfWeek: number[];
  startTime: string;
  endTime: string;
  price: number;
  invalidReason: string | null;
}
